src/main/java/com/svydovets/bibirnate/mapper/RegularFieldMapper.java

```diff
@@ -0,0 +1,71 @@
+package com.svydovets.bibirnate.mapper;
+
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Blob;
+import java.sql.Date;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import com.svydovets.bibirnate.exceptions.EntityMappingException;
+import com.svydovets.bibirnate.exceptions.TypeMismatchException;
+
+
+public class RegularFieldMapper implements EntityFieldMapper {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> void mapField(Field field, T instance, Object value) {
+        try {
+            field.setAccessible(true);
+            value = defineValue(value, field);
+            field.set(instance, value);
+        } catch (IllegalAccessException ex) {
+            throw new EntityMappingException(String.format("Could not access field: %s while mapping entity: %s",
+              field.getName(), instance.getClass().getSimpleName()), ex);
+        } catch (IllegalArgumentException ex) {
+            throw new TypeMismatchException(
+              String.format("Database type: %s doesn't match provided entity type: %s",
+                value.getClass().getSimpleName(), field.getType().getSimpleName()), ex);
+        }
+    }
+
+    private Object defineValue(Object value, Field field) {
+        if (value instanceof Integer intVal
+            && (field.getType().equals(Short.class) || field.getType().equals(short.class))) {
+            value = intVal.shortValue();
+        } else if (value instanceof BigDecimal bdVal && field.getType().equals(BigInteger.class)) {
+            value = bdVal.toBigInteger();
+        } else if (value instanceof Timestamp timestamp) {
+            value = setTimestampField(field, timestamp);
+        } else if (value instanceof Blob blob) {
+            try {
+                var len = blob.length();
+                value = blob.getBytes(1, (int) len);
+            } catch (SQLException ex) {
+                throw new EntityMappingException("Error reading Blob from database", ex);
+            }
+        } else if (value instanceof String strVal
+            && (field.getType().equals(Character.class) || field.getType().equals(char.class))) {
+            value = strVal.charAt(0);
+        }
```

Comment:
Maybe we should add else and throw UnsupportedTypeException?

Reply:
Here's a bit different logic. 
When the application is in mapField() method - this means that in previous steps a correct mapper was selected - in current case - RegularFieldMapper. And in such a case we have to support all possible data types.
If the field type is not supported - we'll get an exception while choosing mapper.